processing/src/main/java/org/apache/druid/segment/column/SimpleColumnHolder.java

```diff
@@ -61,6 +63,8 @@
           capabilities.getType() == ValueType.COMPLEX,
           "Only complex column types can have nullable column suppliers"
       );
+      log.warn("Null column supplier for a complex column was passed which indicates "
```

Comment:
Sorry I just looked even closer, I now think the _actual_ best place for this log is probably in `ComplexColumnPartSerde`, in [the `getDeserializer` method](https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/segment/serde/ComplexColumnPartSerde.java#L73) if serde is null. This way you can include the 'typeName' of the missing type, which isn't available in `SimpleColumnHolder`, my bad.

Reply:
Ah, yes. Would the better place to log this warning be then in the [constructor](https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/segment/serde/ComplexColumnPartSerde.java#L40) itself since that is the first place where `serde` gets set to null?

Reply:
Either place seems fine to me